.travis.yml

```diff
@@ -1,19 +1,56 @@
 language: java
 sudo: required
 dist: trusty
+
+env:
+  matrix:
+    - ENGINE=lucee@4.5
+    - ENGINE=lucee@5
+    - ENGINE=adobe@10
+    - ENGINE=adobe@11
+    - ENGINE=adobe@2016
+
+cache:
+  directories:
+  - $HOME/.CommandBox
+
 before_install:
-- sudo apt-key adv --keyserver keys.gnupg.net --recv 6DA70622
-- sudo echo "deb http://downloads.ortussolutions.com/debs/noarch /" | sudo tee -a
-  /etc/apt/sources.list.d/commandbox.list
+  - sudo apt-key adv --keyserver keys.gnupg.net --recv 6DA70622
+  - sudo echo "deb http://downloads.ortussolutions.com/debs/noarch /" | sudo tee -a
+    /etc/apt/sources.list.d/commandbox.list
+
 install:
-- sudo apt-get update && sudo apt-get --assume-yes install commandbox
-- cd tests
-- box install
-- box server start
+  # Install CommandBox and our JSON parsing binary
+  - sudo apt-get update && sudo apt-get --assume-yes install commandbox jq
+  # Test that the box binary is available and ready for our tests
+  - box version
+  # If using auto-publish, you will need to provide your API token with this line:
+  - box config set endpoints.forgebox.APIToken=$FORGEBOX_API_TOKEN > /dev/null
+
 before_script:
-# Startup the app
-- curl http://localhost:49615/
-script: > 
-    testResults="echo $(box testbox run)";
-    echo "$testResults";
-    if grep -i "\[Failures: [1-9][0-9]\?[0-9]\?\]\|\[Errors: [1-9][0-9]\?[0-9]\?\]\|<t[^>]*>\|<b[^>]*>" <<< $testResults;  then exit 1; fi
\ No newline at end of file
+  # Install our test dependencies
+  - cd $TRAVIS_BUILD_DIR/tests && box install && cd $TRAVIS_BUILD_DIR
+  # Set our matrix engine
+  - box server set app.cfengine=$ENGINE
+  # Start our server for tests
+  - box server start
+
+script: 
+  # Run our tests - CommandBox will return a non-zero exit code if they fail, which will fail the build
+  - box testbox run runner=http://127.0.0.1:49615/tests/runner.cfm
+
+# Deploy on after_success, which saves us from writing a separate bash script for deployment
+after_success:
+  # Evaluate our version for the release
+  - if [ $TRAVIS_BRANCH != 'master' ]; then VERSION_APPENDIX='-snapshot'; fi 
+  - MODULE_VERSION="$(cat box.json | jq -r '.version')$VERSION_APPENDIX"
+  - echo "Module version set to $MODULE_VERSION"
+  # Assign our location archive to the commit
+  - box package set location="https://github.com/coldbox-modules/cbox-bcrypt/archive/$TRAVIS_COMMIT.zip"
+  # Debug our box json file
+  - cat box.json
+  # Stable Releases
+  - if [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ $TRAVIS_BRANCH == 'master' ] && [ $ENGINE == 'lucee@4.5' ]; then box bump version="$MODULE_VERSION+$TRAVIS_BUILD_NUMBER" --force; fi
+  # Unstable Releases
+  - if [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ $TRAVIS_BRANCH != 'master' ] && [ $ENGINE == 'lucee@4.5' ]; then box bump version="$MODULE_VERSION" --minor --force; fi
+  
```

Comment:
Since we have adopted this into the Ortus Fold, we need to do the same as the unified workbench:

- Create API Docs + upload them to s3 under `coldbox-modules/bcrypt`
- Upload binaries to s3 downloads using `coldbox-modules/bcrypt`